src/Cohesity.Powershell.Models/AccessTokenCredential.cs

```diff
@@ -21,38 +21,88 @@ namespace Cohesity.Model
+        /// <value>Specifies OTP type for MFA verification. &#39;Totp&#39; implies the code is TOTP. &#39;Email&#39; implies the code is email OTP.</value>
+        [JsonConverter(typeof(StringEnumConverter))]
+        public enum OtpTypeEnum
+        {
+            /// <summary>
+            /// Enum Totp for value: Totp
+            /// </summary>
+            [EnumMember(Value = "Totp")]
+            Totp = 1,
+
+            /// <summary>
+            /// Enum Email for value: Email
+            /// </summary>
+            [EnumMember(Value = "Email")]
+            Email = 2
+
+        }
+
+        /// <summary>
+        /// Specifies OTP type for MFA verification. &#39;Totp&#39; implies the code is TOTP. &#39;Email&#39; implies the code is email OTP.
         /// </summary>
-        [JsonConstructorAttribute]
-        protected AccessTokenCredential() { }
+        /// <value>Specifies OTP type for MFA verification. &#39;Totp&#39; implies the code is TOTP. &#39;Email&#39; implies the code is email OTP.</value>
+        [DataMember(Name="otpType", EmitDefaultValue=true)]
+        public OtpTypeEnum? OtpType { get; set; }
         /// <summary>
         /// Initializes a new instance of the <see cref="AccessTokenCredential" /> class.
         /// </summary>
+        /// <param name="certificate">Specifies the certificate for logging in the cert base auth cluster..</param>
         /// <param name="domain">Specifies the domain the user is logging in to. For a Local user model, the domain is always LOCAL. For LDAP/AD user models, the domain will map to an LDAP connection string. A user is uniquely identified by a combination of username and domain. If this is not set, LOCAL is assumed..</param>
-        /// <param name="password">Specifies the password of the Cohesity user account. (required).</param>
-        /// <param name="username">Specifies the login name of the Cohesity user. (required).</param>
-        public AccessTokenCredential(string domain = default(string), string password = default(string), string username = default(string))
+        /// <param name="otpCode">Specifies OTP code for MFA verification..</param>
+        /// <param name="otpType">Specifies OTP type for MFA verification. &#39;Totp&#39; implies the code is TOTP. &#39;Email&#39; implies the code is email OTP..</param>
+        /// <param name="password">Specifies the password of the Cohesity user account..</param>
+        /// <param name="privateKey">Specifies the matching private key of the above certificate..</param>
+        /// <param name="username">Specifies the login name of the Cohesity user..</param>
+        public AccessTokenCredential(string certificate = default(string), string domain = default(string), string otpCode = default(string), OtpTypeEnum? otpType = default(OtpTypeEnum?), string password = default(string), string privateKey = default(string), string username = default(string))
         {
+            this.Certificate = certificate;
```

Comment:
Should we leave certificate/private key out of this ?

Reply:
I created this model with latest yaml file, to include otpType and otpcode in this model. This includes all other variables as well.

Reply:
Ack, lets keep iit.

Reply:
@KavishreeShanmugam11 do you know when 1.8.1 is going for build?  I'm waiting for the fix to Get-CohesityProtectionSource (I have it locally) as I need to push the API through our release pipeline